test/orbit_web/controllers/auth_controller_test.exs

```diff
@@ -0,0 +1,53 @@
+defmodule OrbitWeb.AuthControllerTest do
```

Comment:
A case that's not covered here: If you're already logged in, and you try to visit `/login`, it redirects you to `/`.

Reply:
This is annoying to implement, as data about being logged in is inserted by the `RequireLogin` plug, but we don't put `/login` behind that plug (for obvious reasons). This doesn't work in Glides either.

Reply:
🤯 I didn't realize that was broken on Glides.

I guess this line in `auth_controller.ex` doesn't do anything then.
https://github.com/mbta/orbit/pull/21/files#diff-aa3dda579650ee9ac3c145b3e3254d8dddac9fb940e94016013ea10123fc414aR8

```ex
  def login_page(conn, _params) do
    if Auth.logged_in_user(conn) do
      conn
      |> redirect_login_successful()
```

(I'm not asking you to delete it though.)

Reply:
Perhaps we could instead pipe the login path through the [`Guardian.Plug.LoadResource`](https://hexdocs.pm/guardian/Guardian.Plug.LoadResource.html#content) plus with `allow_blank: true`? I think that would make the user information available if present, and otherwise not cause the request to fail.

Reply:
Hmm. We're already using `allow_blank` in `OrbitWeb.Auth.Guardian.Pipeline`, and I'm not sure what using LoadResource's `allow_blank` _without_ `VerifySession` does from a security standpoint. Perhaps I'll just pull this functionality from `login_page/2` and log a task?

Because /login on Orbit is an immediate redirect to `/auth/keycloak`, it has the benefit of looking like a redirect to `/` even if slow.

Reply:
My understanding is that `LoadResource` with `allow_blank: true` and none of the other plugs like `VerifySession` or `VerifyHeader` would load the resource, if present, and otherwise do nothing. That would give us access to the user resource without necessarily implying that the user is correctly logged in. That said, as I think about this more I'm not sure how we would avoid potential redirect loops if we found a logged in user but then the page we redirected them to redirected right back to the login flow because the login isn't actually valid. All of that said, I think I'd be fine with removing that code from the PR for now and creating a followup ticket; it's not super-critical functionality.